pkg/eventpersister/persister/persister.go

```diff
@@ -688,3 +688,43 @@ func (p *Persister) getUserEvaluation(
 	}
 	return evaluation, nil
 }
+
+func (p *Persister) setExpiration(dCmd *goredis.DurationCmd, key string) error {
+	// The value of command is available only after the pipeline is executed.
+	// https://redis.uptrace.dev/guide/go-redis-pipelines.html#pipelines
+	d, err := dCmd.Result()
+	if err != nil {
+		return err
+	}
+	if d == -1 {
+		// The expiration may be overriden because of race condition.
+		// However, we don't have to care it because this expiration is not have to be strict.
+		_, err := p.evaluationCountCacher.Expire(key, oneMonth)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (p *Persister) countEvent(key string) error {
+	pipe := p.evaluationCountCacher.Pipeline()
+	pipe.Incr(key)
+	dCmd := pipe.TTL(key)
```

Comment:
I understand that the go-redis only returns the incremented value after the pipe is executed. But using the `pipe.TTL`, won't change the issue of calling two operations every time we increment a key.

I think we can only achieve this using the Lua script.
We can simplify all this using the example below.

E.g

```go
var incr = goredis.NewScript(`
local key = KEYS[1]
local val = redis.call("INCR", key)
if val == 1 then
  redis.call('EXPIRE', key, ARGV[1])
end
return v
`)
keys := []string{"key"}
values := []interface{}{1670819943}
_, err := incr.Run(ctx, rdb, keys, values...).Int()
```

Reference:
https://redis.uptrace.dev/guide/lua-scripting.html#redis-script